src/main/java/com/zendesk/maxwell/schema/columndef/DateTimeColumnDef.java

```diff
@@ -19,7 +23,24 @@ public static DateTimeColumnDef create(String name, String type, short pos, Long
 
 	protected String formatValue(Object value, MaxwellOutputConfig config) throws ColumnDefCastException {
 		// special case for those broken mysql dates.
-		if ( value instanceof Long ) {
+		if ( value instanceof String) {
+			String dateString = (String) value;
+
+			if ( "0000-00-00 00:00:00".equals(dateString) ) {
+				if ( config.zeroDatesAsNull )
+					return null;
+				else 
+					return appendFractionalSeconds("0000-00-00 00:00:00", 0, getColumnLength());
+			} else {
+				if ( !DateValidator.isValidDateTime(dateString) )
```

Comment:
sorry for the delay, I'm coming back to this PR since I'm prepping a release of your mariadb stuff and some other stuff too.  I'm a little worried about running a regex in a very hot code-path...

But I guess this only runs on bootstrapping so it might be ok?  can you confirm that?